projects/igniteui-angular/src/lib/data-operations/expressions-tree-util.ts

```diff
@@ -149,7 +149,7 @@ function recreateExpression(expression: IFilteringExpression, fields: FieldType[
         }
     }
 
-    if (!expression.condition && expression.conditionName) {
+    if (!expression.condition && !expression.conditionName) {
```

Comment:
This error appears when rehydration has failed. If a `conditionName`  is present and the `condition` has not been populated - it should throw.
This change would likely make the code silently fail when rehydration is unsuccessful and be challenging to debug.

Reply:
Most cases where rehydration would fail should be coverred in https://github.com/IgniteUI/igniteui-angular/pull/15506.

Reply:
@Otixa Why it would throw an error when it is supposed to work without `condition` set?

Reply:
Hm I will check again to see the expression I tested again, might be related to that why `condition` was invalid in this case.

Reply:
Lastly it seems like the linked PR resolves most of the issues. I've added only a few safety check if the columns array is an empty string in this case.